core/lib/zksync_core/src/api_server/web3/namespaces/snapshots.rs

```diff
@@ -0,0 +1,78 @@
+use crate::api_server::web3::state::RpcState;
+use crate::l1_gas_price::L1GasPriceProvider;
+use zksync_types::snapshots::{AllSnapshots, Snapshot, SnapshotChunkMetadata, SnapshotStorageKey};
+use zksync_types::L1BatchNumber;
+use zksync_web3_decl::error::Web3Error;
+
+#[derive(Debug)]
+pub struct SnapshotsNamespace<G> {
+    pub state: RpcState<G>,
+}
+
+impl<G> Clone for SnapshotsNamespace<G> {
+    fn clone(&self) -> Self {
+        Self {
+            state: self.state.clone(),
+        }
+    }
+}
+impl<G: L1GasPriceProvider> SnapshotsNamespace<G> {
+    pub fn new(state: RpcState<G>) -> Self {
+        Self { state }
+    }
+    pub async fn get_all_snapshots_impl(&self) -> Result<AllSnapshots, Web3Error> {
+        let mut storage_processor = self.state.connection_pool.access_storage().await.unwrap();
+        let mut snapshots_dal = storage_processor.snapshots_dal();
+        Ok(snapshots_dal.get_all_snapshots().await.unwrap())
+    }
+
+    pub async fn get_snapshot_by_l1_batch_number_impl(
+        &self,
+        l1_batch_number: L1BatchNumber,
+    ) -> Result<Option<Snapshot>, Web3Error> {
+        let mut storage_processor = self.state.connection_pool.access_storage().await.unwrap();
+        let mut snapshots_dal = storage_processor.snapshots_dal();
+        let snapshot_files = snapshots_dal
+            .get_snapshot_files(l1_batch_number)
+            .await
+            .unwrap();
```

Comment:
Errors should be propagated here using [`internal_error()`](https://github.com/matter-labs/zksync-era/blob/40328a3a195b3544d50dec723eeff3b6737cd3e8/core/lib/zksync_core/src/api_server/web3/backend_jsonrpc/error.rs#L39).